drivers/hwinfo/hwinfo_gecko.c

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2021 Sun Amar
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <em_system.h>
+#include <drivers/hwinfo.h>
+#include <string.h>
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+{
+	uint64_t unique_id = SYSTEM_GetUnique();
```

Comment:
I think you need to convert to big endian first.

Reply:
yea I was not sure about thar but you are probably right. 
I've fixed it